packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -3128,19 +3110,107 @@ export class StripeHelper extends StripeHelperBase {
 
     const productPaymentCycleOld = this.stripePlanToPaymentCycle(planOld);
 
+    const sameBillingCycle =
+      productPaymentCycleOld === baseDetails.productPaymentCycleNew;
+
+    // get next invoice details
+    const nextInvoice = upcomingInvoiceWithInvoiceItem
+      ? upcomingInvoiceWithInvoiceItem
+      : await this.previewInvoiceBySubscriptionId({
+          subscriptionId: subscription.id,
+        });
+
+    const {
+      total: nextInvoiceTotal,
+      currency: nextInvoiceCurrency,
+      lines: nextInvoiceLines,
+    } = nextInvoice || {};
+
+    // https://stripe.com/docs/api/invoices/object#invoice_object-lines
+    // as per the Stripe docs, one of the individual line items includes
+    // new plan pricing and tax information without additional fees
+    const upcomingInvoiceLineItem =
+      nextInvoiceLines && nextInvoiceLines.data
+        ? nextInvoiceLines.data.find((line) => !line.proration)
+        : undefined;
+
+    const {
+      amount: upcomingInvoiceAmount,
+      discount_amounts: upcomingInvoiceDiscountAmounts,
+      tax_amounts: upcomingInvoiceTaxAmounts,
+    } = upcomingInvoiceLineItem || {};
+
+    const sameBillingCondition =
+      sameBillingCycle && upcomingInvoiceLineItem !== undefined;
+
+    const newListPriceTotalWithoutTax =
+      sameBillingCondition && upcomingInvoiceAmount !== undefined
+        ? upcomingInvoiceAmount
+        : undefined;
+
+    const newListPriceDiscounts =
+      sameBillingCondition &&
+      upcomingInvoiceDiscountAmounts !== null &&
+      upcomingInvoiceDiscountAmounts !== undefined
+        ? upcomingInvoiceDiscountAmounts.reduce(
+            (acc, discount) =>
+              discount.amount ? acc + discount.amount : acc + 0,
+            0
+          )
+        : 0;
```

Comment:
This could be refactored to the following

```
const newListPriceDiscounts =
      sameBillingCondition &&
      upcomingInvoiceDiscountAmounts?.reduce(
            (acc, discount) =>
              discount.amount ? acc + discount.amount : acc,
            0
          )
        || 0;
```